r2dbc/boot-r2dbc-json-column/src/main/java/com/example/learning/handler/PostHandler.java

```diff
@@ -24,15 +36,30 @@ public PostHandler(PostRepository postRepository, CommentRepository commentRepos
     }
 
     public Mono<ServerResponse> all(ServerRequest req) {
-        return ok().body(
-                        this.postRepository.findAll().flatMap(post -> commentRepository
-                                .findByPostId(post.getId())
-                                .collectList()
-                                .map(comments -> {
-                                    post.setComments(comments);
-                                    return post;
-                                })),
-                        Post.class);
+        int page = Integer.parseInt(req.queryParam("page").orElse("0"));
+        int size = Integer.parseInt(req.queryParam("size").orElse("10"));
```

Comment:
@coderabbitai, how to add validation to both page and size that they are non-negative if present and also implement sorting based on request

Reply:
@coderabbitai, above code is not throwing a bad request, if user has inputted negative numbers, please throw bad request